### ISSUE TITLE
Add DOORBELL to binary_sensor DEVICE_CLASS

### ISSUE THREAD

**Issue by ochlocracy** [home-assistant/architecture#302]
## Context

Would like to add `DEVICE_CLASS_DOORBELL` to the `binary_sensor`.

While completing implementation of the [`Alexa.DoorbellEventSource`](https://developer.amazon.com/docs/device-apis/alexa-doorbelleventsource.html) Interface for the `alexa` integration, I struggled to come up with an eloquent way to filter entities to expose this interface for. 

Using the current model, I only had the logical options of `DEVICE_CLASS_OCCUPANCY`, `DEVICE_CLASS_PRESENCE`, or `DEVICE_CLASS_MOTION`. None of those fit 100% and would yield the `Alexa.DoorbellEventSource` Interface to any of these sensors.

Also, attempted to hack together a template binary_sensor, to work around this, but ran into the same issues.

I researched the doorbell integrations; and most support the doorbell as a binary_sensor, with device class of `occupancy`. Adding `doorbell` would better classify these sensors. 

## Proposal

```python
DEVICE_CLASS_DOORBELL = "doorbell"
CONF_IS_PRESSED = "is_pressed"
CONF_IS_NOT_PRESSED = "is_not_pressed"
CONF_PRESSED = "pressed"
CONF_NOT_PRESSED = "not_pressed"
```

press/push are the two options I thought of for the condition.

## Consequences

I do not foresee any consequences, except missing out on the opportunity to name it `DEVICE_CLASS_DING_DONG`

**Comment by emontnemery** [comment]
I think doorbells, remotes etc. are better suited as just events generated by a device. Have a look at the device triggers recently implemented for zha and deconz.

**Comment by pvizeli** [comment]
I agree with @emontnemery, our model is to send events for non state based actions

**Comment by ochlocracy** [comment]
Acknowledged. 
I'll continue researching to try to work around my current issue. Thanks for the input.

**Comment by dwradcliffe** [comment]
@ochlocracy Did you find any other way to accomplish this without requiring manual override config? I came here to propose the same thing. The only way I can see to send the doorbell event to Alexa is via an entity which currently requires manual override configuration.

Is there some way to expose an entity to Alexa without the entity actually existing? Perhaps the component can translate an event into an entity on the alexa side?

**Comment by ochlocracy** [comment]
@dwradcliffe Nope. 

I investigated watching for an event like suggested, all doorbell integrations are currently integrating the pressing of the bell as a `binary_sensor` instead of firing a specific doorbell type event. You would still need to classify or configure the device/entity you wanted to track though some configuration workaround to let us know "this is a door bell". 

As far as I know there is no way to accurately search the entity registry for anything that is a doorbell, and the pressing of the bell is a state change, and not an event.... But it's been several months, since I looked and I know a lot has changed...

In my personal setup, I use a [`template` `binary_sensor`](https://www.home-assistant.io/integrations/binary_sensor.template/) and then then easily modify the logic that triggers that sensor.
